### DISCUSSION TITLE
Directus JS SDK support for assets

### DISCUSSION THREAD

**Discussion by workatease** [directus/directus#4194] [Archive]
I don't see any documentation for assets in SDK
And idea on how to implement it via SDK

**Comment by philleepflorence** [comment]
Is the Assets Documentation different from the [JavaScript SDK File Reference](https://docs.directus.io/reference/sdk-js/#files)?

Or are you trying to access a file via the [Assets - API Reference](https://docs.directus.io/reference/api/assets/#assets)?

**Reply by workatease** [reply]
Yes ... I am trying to access the assets .. and I thought I needed the SDK to access it.

I was able to access it via the endpoint by passing the access_token as query param.

**Reply by philleepflorence** [reply]
Oh, are you trying to **Get Asset URL** or **Get Asset** (Return the URL of an asset's original or thumbnail) using the SDK?
The current SDK doesn't seem to include that method.

To stream or retrieve the actual file contents via the Assets API Endpoint - [API Reference - Assets - Requesting a Thumbnail](https://docs.directus.io/reference/api/assets/#requesting-a-thumbnail).

**Reply by workatease** [reply]
I am trying to get asset url

**Reply by rijkvanzanten** [reply]
The asset URL is always `/assets/` + the ID of the file. You can read using `sdk.files.read()` and use the IDs from there 👍🏻
